src/main/java/com/diffmin/patch/PatchApplication.java

```diff
@@ -71,8 +72,15 @@ private static void performInsertion(
 
         switch (toBeInserted.getRoleInParent()) {
             case STATEMENT:
-                ((CtStatementList) inWhichElement)
-                        .addStatement(where, (CtStatement) toBeInserted.clone());
+                Optional<CtStatement> statement =
+                        ((CtStatementList) inWhichElement)
+                                .getStatements().stream()
+                                        .filter(st -> st == toBeInserted)
+                                        .findFirst();
+                if (statement.isEmpty()) {
+                    ((CtStatementList) inWhichElement)
+                            .addStatement(where, (CtStatement) toBeInserted.clone());
+                }
```

Comment:
This seems like a dirty workaround, but I don't think it is. Consider the case in https://github.com/GumTreeDiff/gumtree/issues/270. When `TreeNode(int, TreeNode, TreeNode)` is inserted, all the statements are also inserted as its children, including `this.val = val`. However, when apply the move patch of of `this.val = val`, we try to insert it again. Hence, this function prevents that.